fiftyone/core/map/batcher/id_batch.py

```diff
@@ -0,0 +1,72 @@
+"""
+Abstract mapping backend
+
+| Copyright 2017-2025, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+
+import functools
+from typing import List, Optional, TypeVar
+
+import bson
+
+import fiftyone.core.utils as fou
+import fiftyone.core.map.batcher.batch as fomb
+import fiftyone.core.map.batcher.slice_batch as foms
+from fiftyone.core.map.typing import SampleCollection
+
+fov = fou.lazy_import("fiftyone.core.view")
+
+T = TypeVar("T")
+
+
+class SampleIdBatch(fomb.SampleBatch):
+    """Sample batch using ids"""
+
+    @classmethod
+    @functools.cache
+    def get_max_batch_size(cls):
+        """Get max batch size"""
+        return fou.recommend_batch_size_for_value(
+            bson.ObjectId(), max_size=100000
```

Comment:
Why max at 100_000? (The mongo limit is 100Mb)

Reply:
Input query limit is 16MB I think, just like a normal document. 100MB is something else - working set document size or something

Reply:
Yeah I wasn't thinking about passing lists of ids when thinking about batch sizes ☠️